powertools-metrics/src/test/java/software/amazon/lambda/powertools/metrics/MetricsLoggerTest.java

```diff
@@ -124,13 +126,29 @@ void singleMetricsCaptureUtilityWithDefaultNameSpace() {
 
     @Test
     void metricsLoggerCaptureUtilityWithDefaultNameSpace() {
+        testWithNewLogger(MetricsUtils::withMetricsLogger);
+    }
+
+    @Test
+    void deprecatedMetricLoggerCaptureUtilityWithDefaultNameSpace() {
+        testWithNewLogger(MetricsUtils::withMetricLogger);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenDefaultDimensionIsNull() {
+        assertThatNullPointerException()
+                .isThrownBy(() -> MetricsUtils.defaultDimensionSet(null))
+                .withMessage("Null dimension set not allowed");
+    }
+
+    private void testWithNewLogger(Consumer<Consumer<MetricsLogger>> methodToTest) {
```

Comment:
strange to name it `NewLogger` when you use it with the deprecated one...

```suggestion
    private void testLogger(Consumer<Consumer<MetricsLogger>> loggerToTest) {
```

Reply:
New was as opposed to the default `MetricsUtils.metricsLogger()` and `MetricsUtils.withMetric(s)Logger` essentially providing new ones. However, happy to call it `testLogger` instead to avoid any confusion